components/Typography/examples/TypographyMaterialScalableStylesViewController.m

```diff
@@ -0,0 +1,173 @@
+                                               name:UIContentSizeCategoryDidChangeNotification
+                                             object:nil];
+}
+
+- (void)contentSizeCategoryDidChange:(NSNotification *)notification {
+  NSString *sizeCategory = notification.userInfo[UIContentSizeCategoryNewValueKey];
+  NSLog(@"New size category : %@", sizeCategory);
+
+  // Update font array to reflect new size category
+  _styleFonts = @[
+    [_typography.body1 mdc_scaledFontAtDefaultSize],
+    [_typography.body1 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.body2 mdc_scaledFontAtDefaultSize],
+    [_typography.body2 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.caption mdc_scaledFontAtDefaultSize],
+    [_typography.caption mdc_scaledFontForCurrentSizeCategory],
+    [_typography.button mdc_scaledFontAtDefaultSize],
+    [_typography.button mdc_scaledFontForCurrentSizeCategory],
+    [_typography.overline mdc_scaledFontAtDefaultSize],
+    [_typography.overline mdc_scaledFontForCurrentSizeCategory],
+    [_typography.subtitle1 mdc_scaledFontAtDefaultSize],
+    [_typography.subtitle1 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.subtitle2 mdc_scaledFontAtDefaultSize],
+    [_typography.subtitle2 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline1 mdc_scaledFontAtDefaultSize],
+    [_typography.headline1 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline2 mdc_scaledFontAtDefaultSize],
+    [_typography.headline2 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline3 mdc_scaledFontAtDefaultSize],
+    [_typography.headline3 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline4 mdc_scaledFontAtDefaultSize],
+    [_typography.headline4 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline5 mdc_scaledFontAtDefaultSize],
+    [_typography.headline5 mdc_scaledFontForCurrentSizeCategory],
+    [_typography.headline6 mdc_scaledFontAtDefaultSize],
+    [_typography.headline6 mdc_scaledFontForCurrentSizeCategory],
+  ];
+
+  [self.tableView reloadData];
+}
+
+#pragma mark - UITableViewDataSource
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+  return _strings.count;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+  return _styleNames.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView
+         cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+  UITableViewCell *cell = [self.tableView dequeueReusableCellWithIdentifier:@"cell"];
+  if (cell == nil) {
+    cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle
+                                  reuseIdentifier:@"cell"];
+  }
+  cell.textLabel.text = _strings[indexPath.section];
+  cell.textLabel.font = _styleFonts[indexPath.row];
+  cell.textLabel.numberOfLines = 0;
+  cell.textLabel.lineBreakMode = NSLineBreakByWordWrapping;
+
+  if (cell.textLabel.font.pointSize > 100 && indexPath.section == 0) {
```

Comment:
This looks like you're attempting to approximate cells that are unreasonably large for the device. I would suggest comparing the `intrinsicContentSize` of the label against the cell bounds (or tableView bounds) and changing the text if it's beyond some limit.

On an iPad Pro, I would hope these large fonts would still be renderable, otherwise it implies we're carrying around effectively-useless fonts.

Reply:
I want it to be accurately sized so people can see whether it makes sense for them.
This is by design.

Reply:
I'm not sure I understand your comment. My suggestion was to only change the label text if the label doesn't fit within the bounds (or is some fraction larger than the bounds).  Why do you think that would not accurately-size the text?